whois-nrtm4/src/main/java/net/ripe/db/nrtm4/publish/SnapshotFileStreamer.java

```diff
@@ -0,0 +1,50 @@
+package net.ripe.db.nrtm4.publish;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.ripe.db.nrtm4.persist.NrtmDocumentType;
+import net.ripe.db.nrtm4.persist.SnapshotObjectRepository;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+@Service
+public class SnapshotFileStreamer {
+
+    private final SnapshotObjectRepository snapshotObjectRepository;
+
+    SnapshotFileStreamer(
+        final SnapshotObjectRepository snapshotObjectRepository
+    ) {
+        this.snapshotObjectRepository = snapshotObjectRepository;
+    }
+
+    public void writeSnapshotAsJson(
+        final PublishableSnapshotFile snapshotFile,
+        final OutputStream outputStream
+    ) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final JsonGenerator jGenerator = mapper.getFactory().createGenerator(outputStream, JsonEncoding.UTF8);
+        jGenerator.writeStartObject();
+        jGenerator.writeNumberField("nrtm_version", snapshotFile.getNrtmVersion());
+        jGenerator.writeStringField("type", NrtmDocumentType.SNAPSHOT.lowerCaseName());
```

Comment:
This is being used to write to fileSystem correct? I thought we already save json in database and we just need to write to file? Why we are converting to json again?

Reply:
Sorry I think I get it now